src/main/java/com/artipie/YamlSettings.java

```diff
@@ -55,26 +71,61 @@ public Storage storage() throws IOException {
         return new YamlStorageSettings(
             Yaml.createYamlInput(this.content)
                 .readYamlMapping()
-                .yamlMapping("meta")
+                .yamlMapping(YamlSettings.META)
                 .yamlMapping("storage")
         ).storage();
     }
 
     @Override
     public CompletionStage<Authentication> auth() throws IOException {
-        //@checkstyle MethodBodyCommentsCheck (12 lines)
-        // @todo #146:30min Implement this method to obtain authentications: for now
-        //  we have AuthFromEnv, which is available when `credentials.type` equals `env`, and
-        //  AuthFromYaml, which is available when `credentials.type` equals
-        //  `file` and `credentials.path` specified. New `credentials` section in main artipie
-        //  config:
-        //  credentials:
-        //    # let's support `env` and `file` types for now
-        //    type: file
-        //    # file location, storage key, relative to `meta.storage.path`
-        //    path: _credentials.yml
-        //  For details check comments in #146.
-        //  After implementing pass found authentications to slices.
-        throw new NotImplementedException("Not yet implemented");
+        final YamlMapping cred = Yaml.createYamlInput(this.content)
+            .readYamlMapping()
+            .yamlMapping(YamlSettings.META)
+            .yamlMapping("credentials");
+        final CompletionStage<Authentication> res;
+        final String path = "path";
+        if (cred != null && "file".equals(cred.string("type")) && cred.string(path) != null) {
```

Comment:
@olenagerasimova I think if user creates configuration with `file` type credentials but doesn't specify `path`, we should not fall back to env credentials but throw an exception (`return CompletableFuture.failedFuture()`), because it's not a valid scenario, rather administrator mistake.